app/collins/solr/CollinsQueryLanguage.scala

```diff
@@ -27,7 +27,7 @@ case class CQLQuery(select: SolrDocType, where: SolrExpression) {
  * itself into a Solr Query
  */
 sealed trait SolrQueryComponent {
-  protected[solr] def traverseQueryString(): String = traverseQueryString(true)
```

Comment:
It's used in one spot: https://github.com/tumblr/collins/blob/master/app/collins/solr/CollinsQueryLanguage.scala#L140

Expressions that have multiple children are not wrapped in parentheses if they are the top level expression. That being said, it shouldn't change anything to actually wrap them, but i'll open another PR to simplify things by removing the `toplevel` parameter in order to keep this one focused on fixing the query caching.